frontend/src/main/webapp/triplist.js

```diff
@@ -0,0 +1,20 @@
+function tripData() {
+  var trips = [
```

Comment:
Instead of setting trips directly in-line (which is not how you'll do it when you're calling into the API), this is better extracted into its own `fetchTrips` function which returns a `Promise`. Then, when you switch to an asynchronous `fetch`-based implementation, you just have to swap out the promise for a call to `fetch`. The logic here would be unchanged, e.g.:

```javascript
fetchTrips().then(trips => {
  setTripListPage(trips)
});
```

Or more concisely:

```javascript
fetchTrips().then(setTripListPage);
```

Alternatively, use `async`/`await` to make this look imperative:
```javascript
async function fetchAndSetTrips() {
  let trips = await fetchTrips();
  setTripListPage(trips);
}
```

Please see [`fetchPlace`](https://github.com/anfee1/step23-2020/blob/875b1b4e96296cc1c372ae53c030d544ca94804a/frontend/src/main/webapp/view_trip.js#L7) for an example showing how to return a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). Reach out to me or @anfee1 if you have questions about how to do this.

Reply:
Note: returning a promise also lets you test out what the page looks like with slow loading by resolving the promise using `setTimeout`. Anthony can also help you out with that.